pkg/sql/schemachanger/scbuild/build.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package scbuild
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scbuild/internal/scbuildstmts"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scerrors"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/errors"
+)
+
+// Build constructs a new state from an initial state and a statement.
+//
+// The function takes an AST for a DDL statement and constructs targets
+// which represent schema changes to be performed.
+func Build(
+	ctx context.Context, dependencies Dependencies, initial scpb.State, n tree.Statement,
+) (scpb.State, error) {
+	bs := newBuilderState(dependencies, initial, n)
+	{
+		bCtx := buildCtx{
+			Dependencies: dependencies,
+			BuilderState: &bs,
+		}
+		if err := ProcessStatement(ctx, bCtx, n); err != nil {
+			return scpb.State{}, err
+		}
+	}
+	return *bs.output, nil
+}
+
+// Export dependency interfaces.
+// These are defined in the scbuildstmts package instead of scbuild to avoid
+// circular import dependencies.
+type (
+	// Dependencies contains all the dependencies required by the builder.
+	Dependencies = scbuildstmts.Dependencies
+
+	// CatalogReader contains all catalog operations required by the builder.
+	CatalogReader = scbuildstmts.CatalogReader
+
+	// AuthorizationAccessor contains all privilege checking operations required
+	// by the builder.
+	AuthorizationAccessor = scbuildstmts.AuthorizationAccessor
+)
+
+// ProcessStatement is its own public function to facilitate testing.
```

Comment:
why define this here and not in `scbuildstmts`? I don't see a big motivation for exporting each individual statement from that package. If you did that, I'd just call this `Process` and maybe as an extra nit, lose the final `s` in that package name so this entrypoint would become `scbuildstmt.Process(ctx, bCtx, stmt)`

Reply:
I... hadn't thought of that! Let's do it.